### ISSUE TITLE
Improving error description in BadValueError

### ISSUE THREAD

**Issue by ewoudwerkman** [pyecore/pyecore#61]
If you do a wrong assignment of the specified datatype of an attribute, you get a BadValueError. In most of the errors they are described as follows:
```
pyecore.valuecontainer.BadValueError: Expected type 
      <pyecore.ecore.EProxy object at 0x05EA6130>, but got type int with value 0 instead
```
As you can see it only shows the EProxy and not the datatype the EProxy is proxying. My suggestion is to change the following:
```python
class BadValueError(TypeError):
    def __init__(self, got=None, expected=None):
        msg = "Expected type {0}, but got type {1} with value {2} instead"
        msg = msg.format(expected, type(got).__name__, got)
        super().__init__(msg)
```
in 
```python
msg = msg.format(expected.name, type(got).__name__, got)
```
assuming that ```expected``` is always ```EClass```-based (and the ```name``` attribute is available). Now you get errors like:
```
pyecore.valuecontainer.BadValueError: Expected type EDouble, but got type int with value 0 instead
```
Am I correct?

**Comment by ewoudwerkman** [comment]
Yes, the feature name is helpful for debugging. But that would change the parameters of the ```BadValueError```, adding the feature's name. As far as I can see ```BadValueError``` is not used anywhere else besides the call in ```check()```, and there you can easily use ```self.feature.name```. You could even add ```self.owner.eClass.name``` so you also know which class has this attribute...

When are you planning a new release, as this would be a welcome addition for the people in my team?

**Comment by aranega** [comment]
@ewoudwerkman I will try to propose a new minor version for the end of the week :) (sorry, I have a lot of constraints lately)

**Comment by aranega** [comment]
@ewoudwerkman Sorry it took me so long. I was not happy with some modifications I made on the exception, so I just changed them and release a new version. The message is now improved (it gives you the type, resolve the proxy if required, the feature and the class the feature is contained in).
This new version also give you accesss to the elements in the exception if you prefer to catch if and to raise a new one or to deal with a better error message. Tell me if it's better for you!
